src/cleanlab_codex/internal/validator.py

```diff
@@ -108,3 +132,38 @@ def is_bad(metric: str) -> bool:
     if is_bad("trustworthiness"):
         return "hallucination"
     return "other_issues"
+
+
+def validate_messages(messages: Optional[list[dict[str, Any]]] = None) -> None:
+    """Validate the format of messages based on OpenAI."""
+
+    if messages is None:
+        return
+    if not isinstance(messages, list):
+        raise TypeError("Messages must be a list of dictionaries.")  # noqa: TRY003
+    for message in messages:
+        if not isinstance(message, dict) or "role" not in message or "content" not in message:
+            raise TypeError("Each message must be a dictionary containing 'role' and 'content' keys.")  # noqa: TRY003
+        if not isinstance(message["content"], str):
+            raise TypeError("Message content must be a string.")  # noqa: TRY003
+
+
+def prompt_tlm_for_rewrite_query(query: str, messages: list[dict[str, Any]], tlm: TLM) -> TLMResponse:
+    """Given the query and message history, prompt the TLM for a response that could possibly be self contained.
+    If the tlm call fails, then the original query is returned.
+    """
+
+    messages_str = ""
+    for message in messages:
+        messages_str += f"{message['role']}: {message['content']}\n"
```

Comment:
Nit. Just use the list comprehension to join the different substrings instead of leaving an extra whitespace for the last entry.

```suggestion
    messages_str = "\n".join([f"{m['role']}: {m['content']}" for m in messages])
```